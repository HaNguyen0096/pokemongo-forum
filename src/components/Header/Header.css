*{
    box-sizing: border-box;
}

.headerTop {
    display: block;
}

img{
    background-size: cover;
    width: 100%;
}

.loginNav{
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: #aaa;
    font-size: 30px;
}

.title{
    display: fix;
    text-align: left;
    margin-left: 10px;
    font-size: 20px;
}

.login-button{
    margin-right: 10px;
}

.TopicLists{
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: #ccc;
    font-size: 30px;
    list-style: none;
}
