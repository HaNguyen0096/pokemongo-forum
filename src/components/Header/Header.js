import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import banner from './go-banner.png'
import TokenService from '../../services/token-service'
import './Header.css'

export default class Header extends Component{
  handleLogoutClick = () => {
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link style={{ textDecoration: 'none' }}
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link className='register-button'
          to='/register'>
          Register
        </Link>
        {' - '}
        <Link className='login-button'
          to='/login'>
          Log in
        </Link>
      </div>
    )
  }

  render(){
    
    return (
      <div className='header'>
        <div className='headerTop'>
          <img src={banner} className="banner" alt="Pokemon GO forum" />
          <div className="loginNav">
            <div className="title">Pokemon GO Forum</div>
            <div className='login-button'>
            {TokenService.hasAuthToken()
              ? this.renderLogoutLink()
              : this.renderLoginLink()}
            </div>
          </div>
        </div>
        <nav className='navBar'>
        </nav>
      </div>
    )
  }
}