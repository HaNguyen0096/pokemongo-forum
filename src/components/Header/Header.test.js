import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Header from './Header'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Header component', () => {

  it('renders Header without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <Header />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for Header as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <Header />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders Header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})