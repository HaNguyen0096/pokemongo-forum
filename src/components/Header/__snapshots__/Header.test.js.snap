// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header by default 1`] = `
<div
  className="header"
>
  <div
    className="headerTop"
  >
    <img
      alt="Pokemon GO forum"
      className="banner"
      src="go-banner.png"
    />
    <div
      className="loginNav"
    >
      <div
        className="title"
      >
        Pokemon GO Forum
      </div>
    </div>
  </div>
  <nav
    className="navBar"
  />
</div>
`;

exports[`Header component renders the UI for Header as expected 1`] = `
<div
  className="header"
>
  <div
    className="headerTop"
  >
    <img
      alt="Pokemon GO forum"
      className="banner"
      src="go-banner.png"
    />
    <div
      className="loginNav"
    >
      <div
        className="title"
      >
        Pokemon GO Forum
      </div>
    </div>
  </div>
  <nav
    className="navBar"
  />
</div>
`;
